src/toil/job.py

```diff
@@ -41,55 +42,54 @@
 logger = logging.getLogger( __name__ )
 
 
-class Job(object):
+class ResourceRequirementMixin(object):
     """
-    Class represents a unit of work in toil.
+    Inherit from this class to add resource requirements for an object
     """
-    def __init__(self, memory=None, cores=None, disk=None, preemptable=None, checkpoint=False):
+    def __init__(self, memory=None, cores=None, disk=None):
```

Comment:
So IIRC, preemptability is also a requirement. I think there is even a defaultPreemptable. We should add that to the mixin.